src/ucm/cuda/cudamem.c

```diff
@@ -66,6 +66,12 @@ UCM_OVERRIDE_FUNC(cudaHostUnregister,        cudaError_t)
 
 static void ucm_cuda_set_ptr_attr(CUdeviceptr dptr)
 {
+    if ((void*)dptr == NULL)
+    {
+        ucm_trace("skipping cuPointerSetAttribute for null pointer");
```

Comment:
> It is, and it returns `CUDA_SUCCESS`, for example:

ok, I see
thank you!

Reply:
should we also check cudaMalloc return status here? or it's enough to just check the ptr?

Reply:
The status of `cudaMalloc` is already checked appropriately in https://github.com/openucx/ucx/blob/master/src/ucm/cuda/cudamem.c#L160-L166, before calling `ucm_cuda_set_ptr_attr`. But the call succeeds, since allocating 0 bytes is valid, but passing that null pointer to `cuPointerSetAttribute` isn't.

Reply:
seems to me we are calling `ucm_cuda_set_ptr_attr` also in case of allocation failure

Reply:
You're totally right, I missed that. The newest commit fixes that, thanks for noticing!